src/storage/browser.ts

```diff
@@ -112,7 +113,7 @@ export class BrowserStorage implements IDriveStorage {
             this.buildKey(BrowserStorage.DRIVES_KEY, id)
         );
         if (!drive) {
-            throw new Error(`Drive with id ${id} not found`);
+            throw new DriveNotFoundError(id);
```

Comment:
Other idea I explored in the ReadMode implementation, was to include the errors in the return type of the methods. Since Typescript doesn't have a way to declare what errors a method might throw, it's easy for errors to  only get caught at the root error handling, like in an ErrorBoundary in React. This makes the errors lose context.

In this case, it would return `Promise<Drive | DriveNotFoundError>` instead of throwing errors.
This way, the caller of these method is forced to handle the error case, making it easier to do error handling at the right context level.